src/authkit-callback-route.ts

```diff
@@ -29,7 +29,19 @@ export function handleAuth(options: HandleAuthOptions = {}) {
         url.searchParams.delete('state');
 
         // Redirect to the requested path and store the session
-        url.pathname = returnPathname ?? returnPathnameOption;
+        returnPathname = returnPathname ?? returnPathnameOption;
+
+        // Extract the search params if they are present
+        if (returnPathname.includes('?')) {
+          const newUrl = new URL(returnPathname, 'https://example.com');
+          url.pathname = newUrl.pathname;
+
+          for (const [key, value] of newUrl.searchParams) {
+            url.searchParams.append(key, value);
```

Comment:
have you tested this? normally `url.searchParams` is [readonly](https://developer.mozilla.org/en-US/docs/Web/API/URL/searchParams). (going to approve assuming you have verified this).

Reply:
(i think i'm just reading the docs wrong, this looks good :+1:)

Reply:
Yup works fine in my testing. Also works when goofing around in the chrome console:

![CleanShot 2024-07-22 at 20 37 55](https://github.com/user-attachments/assets/5ae70d7e-7511-472c-a506-66df83c3dcc6)

`url.searchParams` is read-only, meaning `url.searchParams = 'foo'` does nothing. However it has a valid `append` method to add stuff to it.